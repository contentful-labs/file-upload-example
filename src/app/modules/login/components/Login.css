@import '../../../assets/styles/variables.css';

.wrapper {
  display: flex;
  flex-direction: column;
  justify-content: center;
  height: 100%;
}

.group {
  display: flex;
  justify-content: center;
  margin-bottom: var(--layout-gutter);

  & > label {
    flex: 1 1 150px;
    white-space: nowrap;
    text-overflow: ellipsis;
    overflow: hidden;
  }
  & > input {
    flex: 5 1 auto;
  }
}
