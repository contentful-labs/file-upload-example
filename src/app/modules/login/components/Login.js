import { h, Component } from 'preact'
import proptypes from 'proptypes'

import styles from './Login.css'

export default class Login extends Component {
  static propTypes = {
    accessToken: proptypes.string.isRequired,
    spaceId: proptypes.string.isRequired,
    setAccessToken: proptypes.func.isRequired,
    setSpaceId: proptypes.func.isRequired
  }
  constructor (props) {
    super(props)
    this._onAccessTokenChange = this._onAccessTokenChange.bind(this)
    this._onSpaceIdChance = this._onSpaceIdChance.bind(this)
  }
  _onAccessTokenChange (e) {
    this.props.setAccessToken(e.target.value.trim())
  }
  _onSpaceIdChance (e) {
    this.props.setSpaceId(e.target.value.trim())
  }
  render ({ accessToken, spaceId }) {
    return (
      <div>
        <div className={styles.wrapper}>
          <div className={styles.group}>
            <label htmlFor='access-token'>Access Token:</label>
            <input
              id='access-token'
              type='text'
              value={accessToken}
              onChange={this._onAccessTokenChange}
              onInput={this._onAccessTokenChange}
              onKeyUp={this._onAccessTokenChange} />
          </div>
          <div className={styles.group}>
            <label htmlFor='space-id'>Space ID:</label>
            <input
              id='space-id'
              type='text'
              value={spaceId}
              onChange={this._onSpaceIdChance}
              onInput={this._onSpaceIdChance}
              onKeyUp={this._onSpaceIdChance} />
          </div>
        </div>
      </div>
    )
  }
}
