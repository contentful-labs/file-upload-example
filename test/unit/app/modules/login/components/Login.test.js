import { h } from 'preact'
import render from 'preact-render-to-string'

import Login from '../../../../../../src/app/modules/login/components/Login'

test('Login', () => {
  const props = {
    accessToken: 'accessToken',
    spaceId: 'spaceId',
    host: 'host',
    hostUpload: 'hostUpload',
    setAccessToken: jest.fn(),
    setHost: jest.fn(),
    setHostUpload: jest.fn(),
    setSpaceId: jest.fn(),
    initClient: jest.fn()
  }
  const html = render(<Login {...props} />)
  expect(html).toMatchSnapshot()
  expect(props.setAccessToken.mock.calls).toHaveLength(0)
  expect(props.setHost.mock.calls).toHaveLength(0)
  expect(props.setHostUpload.mock.calls).toHaveLength(0)
  expect(props.setSpaceId.mock.calls).toHaveLength(0)
  expect(props.initClient.mock.calls).toHaveLength(0)
})
