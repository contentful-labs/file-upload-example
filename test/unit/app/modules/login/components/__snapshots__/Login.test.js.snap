// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login 1`] = `
<div>
  <form class="wrapper">
    <div class="header">
      <img src="test-file-stub" alt="Contentful" />
      <h1>Direct File Upload Example</h1>
      <p>This is an example showing how to upload files directly to the
        <a href="https://www.contentful.com">Contentful headless CMS</a>. To start using it you need to create an account
        and enter your Content Management API credentials later on.</p>
      <p>This example uses Preact, Redux and Redux Saga. The source code is open
        source and you can find it on
        <a href="https://github.com/contentful-labs/file-upload-example">GitHub</a>.</p>
    </div>
    <div class="group">
      <label htmlFor="access-token">Access Token:</label>
      <input id="access-token" type="text" required value="accessToken"
      />
    </div>
    <div class="group">
      <label htmlFor="space-id">Space ID:</label>
      <input id="space-id" type="text" required value="spaceId"
      />
    </div>
    <div class="reveal">
      <span>Change host</span>
    </div>
    <div class="group">
      <input type="submit" value="Connect" />
    </div>
  </form>
</div>
`;
